### ISSUE TITLE
How to clean up CloudFormation when ArgoCD provisioned applications not deleted

### ISSUE THREAD

**Issue by calliope7** [aws-quickstart/cdk-eks-blueprints#908]
### Describe the feature

ArgoCD Known Issue #1 mentions the need to ensure ArgoCD provisioned applications need to be destroyed by ArgoCD before running a 'CDK destroy'.  This also applies to a 'CDK deploy' where the ArgoCD addon is removed from the script.  However, the main problem for us Cloud Formation/CDK novices is .. what to do when this does happen.  Any chance for a paragraph to help us recover from such a gaff?

### Use Case

Its really just a bit of 'work-around' documentation until this becomes a more robust feature.  The problem is that once this happens it seems very difficult to recover.  

### Proposed Solution

A paragraph of doc to help in the recovery.. Eg. 1) use the AWS console to delete your cluster, 2) somehow blow away your Cloud Formation Template, 3) rebuild the cluster using CDK.

### Acknowledgements

- [X] I may be able to implement this feature request

### CDK version used

2.115.0 (build 58027ee)

### EKS Blueprints Version

1.13.1

### Node.js Version

v20.11.0

### Environment details (OS name and version, etc.)

Windows Powershell

**Comment by shapirov103** [comment]
It makes sense to move forward with this issue. 
There are a few paths - deleting resources, by removing namespace finalizer or just deleting the whole cluster as described in proposed solution. 
The correct approach is 1/ remove bootstrap repo from ArgoCD 2/ remove ArgoCD addon from the cluster. This sequence should work ootb.

**Comment by calliope7** [comment]
Thanks Mikhail. 

Unfortunately if you screw up and Don't delete the ArgoCD apps first then in my experience you cannot delete the cluster.  I was wrong footed because initially I was only trying to update the cluster with no ArgoCD (yes should have thought deeper about it).. which launched me along this path.  In the end I had to hunt down resources and nuke them individually through the console before I was able to delete the cluster and redeploy.  One key gotcha are the ENIs that the lambdas hang onto.  After deleting those lambdas the resources are held for another hour or so and only after that was I able to delete the cluster.  Obviously there is a clear progression to ensure this isn't a problem.. but a needed approach when it does go wrong.  Yes ootb is best but K8s is pretty complex but in order to push this forward I can see that me and my team will need to ramp up our understanding.  Pointers in this regard are always welcome.
